src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1365,8 +1365,7 @@ public DType Add(DName name, DType type)
             {
                 fullType = LazyTypeProvider.GetExpandedType(IsTable);
             }
-
-            Contracts.Assert(!TypeTree.Contains(name));
```

Comment:
>Contracts [](http://example.com/codeflow?start=12&length=9)

When do we hit this?

Reply:
When we try to add an existing field to an existing record / table type. Notice that DTypes are immutable, so we're not actually adding anything. Something like

    var type1 = Parse("![a:n,b:s,c:D]");
    var type2 = type1.Add(new DName("b"), DType.Number);

In release builds (i..e, what we've been using for years), this returns a type equivalent to "![a:n,b:n,c:D]"; in debug builds it asserts. Ideally it should return an error, but at this point this is the _de facto_ implementation.